src/fsharp/tc.fs

```diff
@@ -1876,7 +1876,11 @@ let rec ApplyUnionCaseOrExn (makerForUnionCase,makerForExnTag) m cenv env overal
         let mkf = makerForExnTag(ecref)
         mkf,recdFieldTysOfExnDefRef ecref, [ for f in (recdFieldsOfExnDefRef ecref) -> f.Name ]
 
-    | Item.UnionCase ucinfo ->   
+    | Item.UnionCase(ucinfo,showDeprecated) ->   
+        if showDeprecated then
+            let message = sprintf "The union type for union case '%s' was defined with the RequireQualifiedAccessAttribute. Include the name of the union type ('%s') in the name you are using.'" ucinfo.Name ucinfo.Tycon.DisplayName
```

Comment:
This needs to go in FSComp.txt (so it can be localized by Microsoft into Japanese etc.)  Look around tc.fs and elsewhere for examples.